### ISSUE TITLE
The load_and_save test fails

### ISSUE THREAD

**Issue by Apteryks** [J-F-Liu/lopdf#137]
On a system with many cores (20 logical cores), this test seems to fail most of the time (always?).  Here's what it looks like, when building the package with GNU Guix:

```
starting phase `check'
   Compiling lopdf v0.26.0 (/tmp/guix-build-rust-lopdf-0.26.0.drv-0/lopdf-0.26.0)
    Finished release [optimized] target(s) in 4.67s
     Running target/release/deps/lopdf-f9341d5daaae9471

running 15 tests
test datetime::parse_datetime_seconds_missing ... ok
test datetime::parse_datetime_local ... ok
test datetime::parse_datetime_time_missing ... ok
test datetime::parse_datetime_utc ... ok
test parser::tests::parse_real_number ... ok
test parser::tests::parse_name ... ok
test parser::tests::parse_string ... ok
test writer::save_document ... ok
test parser_aux::load_and_save ... FAILED
test reader::load_short_document ... ok
test parser::tests::parse_content ... ok
test creator::create_document ... ok
test reader::load_many_shallow_brackets ... ok
test reader::load_document ... ok
test reader::load_too_deep_brackets ... ok

failures:

---- parser_aux::load_and_save stdout ----
thread 'parser_aux::load_and_save' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/parser_aux.rs:276:51
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    parser_aux::load_and_save

test result: FAILED. 14 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--lib'
error: in phase 'check': uncaught exception:
%exception #<&invoke-error program: "cargo" arguments: ("test" "--release") exit-status: 101 term-signal: #f stop-signal: #f> 
phase `check' failed after 4.7 seconds
command "cargo" "test" "--release" failed with status 101
builder for `/gnu/store/1qwy7hkx4xjj9c70f843249kyacqskhy-rust-lopdf-0.26.0.drv' failed with exit code 1
build of /gnu/store/1qwy7hkx4xjj9c70f843249kyacqskhy-rust-lopdf-0.26.0.drv failed
```

Curiously, on another system with just 4 cores, the test passed.  Perhaps a race condition?

I've disabled the test for now.

Thank you!

**Comment by jgerrish** [comment]
Your insight was correct Apteryks, it's a race issue.

The create_document test in creator.rs creates a document called test_1_create.pdf
The load_and_save test opens and loads this file, so it needs to exist.
cargo can run tests in parallel, so the file may or may not exist at the time load_and_save runs.

I started a branch fixing it, but it looks like this project hasn't had much activity lately.  I can write a PR if there's interest, or let it go.

**Comment by Apteryks** [comment]
@jgerrish it's always good to have bugs fixed; even if it's not merged by upstream in a timely fashion, distribution can apply the patch on their own :-).

**Comment by J-F-Liu** [comment]
PRs are welcome, thanks

**Comment by Heinenen** [comment]
This issue should be fixed with #151 and can probably be closed.
